test/AlarmClockTest.cpp

```diff
@@ -93,146 +70,90 @@ TEST_F(AlarmClockTest, GetSecSleepTimeInMs) {
 
 TEST_F(AlarmClockTest, microsecondsLessThan500ms) {
    int us = 900;
-   StopWatch testTimer;
-   AlarmClock<microseconds> alerter(us);
+   AlarmClock<microseconds> alerter(us, FakeSleep);
+   EXPECT_FALSE(alerter.Expired());
    WaitForAlarmClockToExpire(alerter);
-   int totalTime = testTimer.ElapsedUs();
-   auto maxTime = us + GetTimingLeeway(microseconds(us));
-   EXPECT_TRUE(us <= totalTime) << "AlarmClock didn't sleep for long enough. Slept for: " << totalTime << " us, should be longer than " << us;
-   EXPECT_TRUE(totalTime <= maxTime) << "AlarmClock took too long to expire. Took " << totalTime << " us. Should be less than " << maxTime;
-   std::cout << "Timeout was set for " << us << " us. Actually slept for " << totalTime << " us. Max timeout: " << maxTime << std::endl;
+   EXPECT_TRUE(alerter.Expired());
```

Comment:
And after this FakeSleep should have been called with 900? Or if you are making multiple calls to FakeSleep, the sum should be 900?